docs/compactor.md

```diff
@@ -0,0 +1,86 @@
+    * Incremental-Snapshot-192375
+* Backup-192789
+    * Full-Snapshot-192789
+    * Incremental-Snapshot-192799
+    * Incremental-Snapshot-192809
+    * Incremental-Snapshot-192819
+
+To solve the problem, proposal is:
+1. ETCDBR will take the first full snapshot after it starts main ETCD Process and snapshotter process. But after taking the first full snapshot, snapshotter will only continue taking delta snapshots but no further full snapshots.
+2. Flatten the directory structure of backup folder. Save all the full snapshots, delta snapshots and compacted snapshots under same directory/prefix. Also, name snapshots after revision number instead of timestamps. Restorer will restore from compacted snapshots that have highest revision number in the name and delta snapshots that have higher revision numbers in name than the comapcted full snapshot.
+
+**Proposed Directory Structure**
+* Backup
+    * Full-Snapshot-revision-1
+    * Incremental-Snapshot-revision-2
+    * Incremental-Snapshot-revision-7
+    * Incremental-Snapshot-revision-13
+    * Incremental-Snapshot-revision-15 (It containes command upto revision 18)
+    * Full-Snapshot-revision-18 (Compacted)
+    * Incremental-Snapshot-revision-19
+    * Incremental-Snapshot-revision-23
+    * Incremental-Snapshot-revision-28
+    * Incremental-Snapshot-revision-33 (It containes command upto revision 38)
+    * Full-Snapshot-revision-38 (Compacted)
+    * Incremental-Snapshot-revision-39
+    * Incremental-Snapshot-revision-42
+    * Incremental-Snapshot-revision-43
+    * Incremental-Snapshot-revision-49
+
+3. A meta-data file might need to be added in the Backup directory to mark which compacted snapshot is ready to be used for restoration. Restorer would study the meta-data file and pickup the ready full snapshot from there and apply all the delta snapshots having revision number greater than that. So, with the backup directory as above, if the content of metadata file is like following:
+
+* Full-Snapshot-revision-1
+* Full-Snapshot-revision-18
+
+Then, during restoration Full-Snapshot-revision-18 and all the delta snaphots from revision number 19 to 49 will be applied. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.** As Full-Snapshot-revision-38 is not present in the metadata file, it would not be considered as ready yet.
+
+Another approach to make sure that restorer only picks up snapshots which are ready, full snapshots can be named with extension `.part` until the snapshot is ready. When the snapshot is ready, `.part` extension can be removed from it's name. Restorer would avoid any file having extension as `.part`. 
+
+So, if Full-Snapshot-revision-38 is not ready in the above example, it can be named as `Full-Snapshot-revision-38.part`. Restorer will apply Full-Snapshot-revision-18 and all the delta snaphots from revision number 19 to 49. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.**
```

Comment:
```suggestion
Another approach to make sure that restorer only picks up snapshots which are ready, full snapshots can be named with extension `.part` until the snapshot is ready. When the snapshot is ready, `.part` extension can be removed from it's name. Restorer would avoid any file having extension as `.part`. 

So, if Full-Snapshot-revision-38 is not ready in the above example, it can be named as `Full-Snapshot-revision-38.part`. Restorer will apply Full-Snapshot-revision-18 and all the delta snaphots from revision number 19 to 49. **Restorer would restore from latest ready full snapshot and all subsequent delta snapshots.**

This option is preferred as it avoids introducing a metadata file. But it would work only if renaming to remove the `.part` extension (upon completion of upload) is an atomic operation in all the supported snapstores. This needs to be evaluated.
```